src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @Test
+    void register_userWithNullLogin_notOk() {
+        User nullLoginUser = new User(null, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(nullLoginUser);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithValidPassword_ok() {
+        User expected = new User(VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        registrationService.register(expected);
+        User actual = storageDao.get(expected.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_userWithInvalidPassword_notOk() {
+        User fiveCharacterPasswordUser = new User(VALID_LOGIN_EXAMPLE,
+                INVALID_PASSWORD_EXAMPLE1,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fiveCharacterPasswordUser);
+        });
+
+        User fourCharacterPasswordUser = new User(VALID_LOGIN_EXAMPLE,
+                INVALID_PASSWORD_EXAMPLE2,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fourCharacterPasswordUser);
+        });
+
+        User oneCharacterPasswordUser = new User(VALID_LOGIN_EXAMPLE,
+                INVALID_PASSWORD_EXAMPLE3,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(oneCharacterPasswordUser);
+        });
+
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithNullPassword_notOk() {
+        User nullPasswordUser = new User(VALID_LOGIN_EXAMPLE, null, MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(nullPasswordUser);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithValidAge_ok() {
+        User expected = new User(VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        registrationService.register(expected);
+        User actual = storageDao.get(expected.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_userWithLessThanMinAge_notOk() {
+        User user = new User(VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, INVALID_AGE_EXAMPLE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
```

Comment:
could be extracted to method and reused

Reply:
Frankly, I have just rethought and figured out that these were redundant because we didn't test the working of the Storage.
In this case, I figured it out correctly, didn't I?